http_queue_service.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type httpClient interface {
+	Do(req *http.Request) (resp *http.Response, err error)
+}
+
+type httpQueue struct {
+	httpClient httpClient
+	endpoint   string
+}
+
+func newHttpQueueService(httpClient httpClient, endpoint string) httpQueue {
```

Comment:
I am not sure that it is actually a "Queue". I understand that you are implementing queue interface, but this is really incidental. If the old code uses the word "queue" to send messages to kafka "topic" which it calls a "queue" - doesn't mean the new code has to continue using misleading names. I am not insisting on this as this may not be a practical change at this time. Just wanted to point out.